superset-frontend/src/explore/components/controls/DndColumnSelectControl/DndColumnSelect.tsx

```diff
@@ -229,6 +271,7 @@ export function DndColumnSelect(props: DndColumnSelectProps) {
             ? openPopover
             : undefined
         }
+        selectAllButton={showSelectAllButton ? <SelectAllButton {...selectAllButtonProps} /> : null}
```

Comment:
```suggestion
        selectAllButton={
          showSelectAllButton ? (
            <Button
              buttonStyle="link" 
              disabled={hasSelectedAll}
              onClick={
                hasSelectedAll
                 ? undefined
                 : () => hasSelectedSome ? deselectAll() : selectAll()
               }
            >
              {hasSelectedSome ? t('Deselect all') : t('Select all')}
            </Button>
          ) : null
        }
```

I think it's possible to directly render the button without all the "mode" conversion. This should make things easier to understand.